## *********************************************************
## 
## File autogenerated for the proc_control package 
## by the dynamic_reconfigure package.
## Please do not edit.
## 
## ********************************************************/

from dynamic_reconfigure.encoding import extract_params

inf = float('inf')

config_description = {'upper': 'DEFAULT', 'lower': 'groups', 'srcline': 233, 'name': 'Default', 'parent': 0, 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'cstate': 'true', 'parentname': 'Default', 'class': 'DEFAULT', 'field': 'default', 'state': True, 'parentclass': '', 'groups': [], 'parameters': [{'srcline': 259, 'description': 'X: The proportional coefficient', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'X_P', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'X: The integration coefficient', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'X_I', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'X: The derivative coefficient', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'X_D', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'X: Maximum integration', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'X_I_LIMIT', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'X: Minimum actuation', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'X_MIN_ACTUATION', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'X: Maximum actuation', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'X_MAX_ACTUATION', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'Y: The proportional coefficient', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Y_P', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'Y: The integration coefficient', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Y_I', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'Y: The derivative coefficient', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Y_D', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'Y: Maximum integration', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Y_I_LIMIT', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'Y: Minimum actuation', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Y_MIN_ACTUATION', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'Y: Maximum actuation', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Y_MAX_ACTUATION', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'Z: The proportional coefficient', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Z_P', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'Z: The integration coefficient', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Z_I', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'Z: The derivative coefficient', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Z_D', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'Z: Maximum integration', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Z_I_LIMIT', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'Z: Minimum actuation', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Z_MIN_ACTUATION', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'Z: Maximum actuation', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Z_MAX_ACTUATION', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'YAW: The proportional coefficient', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'YAW_P', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'YAW: The integration coefficient', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'YAW_I', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'YAW: The derivative coefficient', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'YAW_D', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'YAW: Maximum integration', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'YAW_I_LIMIT', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'YAW: Minimum actuation', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'YAW_MIN_ACTUATION', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'YAW: Maximum actuation', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/indigo/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'YAW_MAX_ACTUATION', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -100.0, 'type': 'double'}], 'type': '', 'id': 0}

min = {}
max = {}
defaults = {}
level = {}
type = {}
all_level = 0

#def extract_params(config):
#    params = []
#    params.extend(config['parameters'])    
#    for group in config['groups']:
#        params.extend(extract_params(group))
#    return params

for param in extract_params(config_description):
    min[param['name']] = param['min']
    max[param['name']] = param['max']
    defaults[param['name']] = param['default']
    level[param['name']] = param['level']
    type[param['name']] = param['type']
    all_level = all_level | param['level']

